add docs for google secret rotation (#1996)
nginx/documentation: content/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md

   | Field                       | Description                  | Note |
   |---------------------------- | ---------------------------- | ---- |

>    | Google Secret ID       | The resource name of the secret in Secret Manager | The resource name must match the format `projects/$PROJECT_ID/secrets/$SECRET_ID/versions/$VERSION`, where `$VERSION` can be a specific version ID (for example, `3`) or the special version ID `latest`. |

Reviewer: ```suggestion
   | Google Secret ID       | The resource name of the secret in Secret Manager | The resource name must match the format `projects/$PROJECT_ID/secrets/$SECRET_ID/versions/$VERSION`, where `$VERSION` can be a specific version ID (for example, `3`), a custom alias, or the special version ID `latest`. |
```

>    | Google Secret ID       | The resource name of the secret in Secret Manager | The resource name must match the format `projects/$PROJECT_ID/secrets/$SECRET_ID/versions/$VERSION` where `$VERSION` can be a specific version number (for example, `3`) or an alias such as `latest`. Using `latest` enables automatic certificate rotation. For more information, see [Rotate a Secret Manager certificate (automatic)](#rotate-a-secret-manager-certificate-automatic). |

Reviewer 2: since `latest` is a special keyword offered by Google Secret Manager that is separate from aliases, I think we need to clarify (and maybe offer some other content about how to manage certs with aliases in these docs, as well?)

```suggestion
   | Google Secret ID       | The resource name of the secret in Secret Manager | The resource name must match the format `projects/$PROJECT_ID/secrets/$SECRET_ID/versions/$VERSION` where `$VERSION` can be a specific version ID (for example, `3`), an alias you've created, or the special version ID `latest`. Using `latest` enables automatic certificate rotation. For more information, see [Rotate a Secret Manager certificate (automatic)](#rotate-a-secret-manager-certificate-automatic). |
```

   | Field                       | Description                  | Note |
   |---------------------------- | ---------------------------- | ---- |

>    | Google Secret ID       | The resource name of the secret in Secret Manager | The resource name must match the format `projects/$PROJECT_ID/secrets/$SECRET_ID/versions/$VERSION` where `$VERSION` can be a specific version number (for example, `3`) or an alias such as `latest`. Using `latest` enables automatic certificate rotation — see [Rotate a Secret Manager certificate (automatic)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/overview.md#rotate-a-secret-manager-certificate-automatic" >}}). |

Reviewer 3: In an earlier comment, @JTorreG removed a couple of em-dashes. People read em-dashes today and think "This doc is generated by AI. Can we trust it?" So I suggest:

```suggestion
   | Google Secret ID       | The resource name of the secret in Secret Manager | The resource name must match the format `projects/$PROJECT_ID/secrets/$SECRET_ID/versions/$VERSION` where `$VERSION` can be a specific version number (for example, `3`) or an alias such as `latest`. Using `latest` enables automatic certificate rotation. For more information, see [Rotate a Secret Manager certificate (automatic)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/overview.md#rotate-a-secret-manager-certificate-automatic" >}}). |
```

Author: this is deeply saddening to me as a former English minor who made em-dashes her personality :(

Reviewer 4: yes same reason I removed them, and they actually added nothing of value to the sentence.

   | File Path               | The secret will be written to this file path so it can be used with NGINX directives such as ssl_certificate or ssl_certificate_key in your NGINX configuration. | The path must be unique within the configuration. |

    {{< /table >}}

{{< call-out "tip" "Enable automatic rotation with latest" >}}

> If you set `$VERSION` to `latest`, NGINXaaS automatically picks up any new secret version you add to Secret Manager without a configuration change. NGINXaaS applies new versions to your deployment within approximately 4 hours. See [Rotate a Secret Manager certificate (automatic)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/overview.md#rotate-a-secret-manager-certificate-automatic" >}}) for details.
> 
> {{< /call-out >}}

Reviewer 4: ```suggestion
{{< call-out "tip" "Enable automatic rotation with latest" >}}
If you set `$VERSION` to `latest`, NGINXaaS automatically picks up any new secret version you add to Secret Manager without a configuration change. NGINXaaS applies new versions to your deployment within approximately 4 hours. See [Rotate a Secret Manager certificate (automatic)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/overview.md#rotate-a-secret-manager-certificate-automatic" >}}) for details.
{{< /call-out >}}
```

Reviewer 3: We're supposed to spell out numbers between 0 and 9. Therefore, I suggest

```suggestion
{{< call-out "tip" "Enable automatic rotation with latest" >}}
If you set `$VERSION` to `latest`, NGINXaaS automatically picks up any new secret version you add to Secret Manager without a configuration change. NGINXaaS applies new versions to your deployment within approximately four hours. See [Rotate a Secret Manager certificate (automatic)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/overview.md#rotate-a-secret-manager-certificate-automatic" >}}) for details.
{{< /call-out >}}
```

- Select the configuration and configuration version created in the last section.
- Select **Update Configuration**.

## Rotate a Secret Manager certificate (automatic)

> If your configuration uses `latest` as the version ID, NGINXaaS fetches the latest secret version on the next scheduled sync. When you add a new secret version in Secret Manager, the next sync picks it up and reloads NGINX.

Reviewer: "Next scheduled sync" feels a bit like an internal detail. I think we should also clarify where the version ID is coming from. What about this instead:

```suggestion
If you set the version ID of your secret to `latest`, NGINXaaS fetches the latest secret version.  When you [add a new secret version in Secret Manager](https://docs.cloud.google.com/secret-manager/docs/add-secret-version#add-a-secret-version), NGINXaaS automatically picks up that version within 4 hours.
```

Author: FYI the Rotation Timeline below will note the four hour window, so I will exclude it here

> If your configuration uses a custom alias as the version ID, NGINXaaS fetches whichever version the alias points to on the next scheduled sync. When you update the alias to point to a different version in Secret Manager, the next sync applies that version.

Reviewer: similar comment:

```suggestion
If you set the version ID of your secret to a custom alias, NGINXaaS fetches the secret version the alias points to. When you [update the alias to point to a different version in Secret Manager](https://docs.cloud.google.com/secret-manager/docs/assign-alias-to-secret-version), NGINXaaS automatically picks up that version within 4 hours.
```

No configuration changes are required in either case.

To ensure certificates are automatically rotated:

1. Generate your new certificate and key.
2. Follow Google's instructions to [add a new secret version in Secret Manager](https://docs.cloud.google.com/secret-manager/docs/add-secret-version#add-a-secret-version).

> 3. When rotation is complete, confirm by checking the **Certificates** list for the new serial number, or by inspecting the certificate at your deployment's endpoint.

Reviewer: These steps only apply to the `latest` case (adding a new secret version instead of referencing a different one). Since we can include the Google instructions in line above, can we remove most of this?

```suggestion
No configuration changes are required in either case. To confirm your deployment is using an updated certificate, check the **Certificates** list for the new serial number or inspect the certificate at your deployment's endpoint.
```

### Rotation timeline

> New secret versions are picked up and applied to your deployment within four hours.

Reviewer: If we say "within 4 hours" above, do we still need this?

```suggestion
```

Author: Hm I think you're right, I'll drop the "Rotation Timeline" heading since it's one line away.

## Rotate a Secret Manager certificate (manual)

To immediately refetch secrets without editing your NGINX configuration, use **Reapply Configuration**. This is useful in the following scenarios:

- **New secret version**: You've uploaded a new certificate and want NGINXaaS to use it right away.
- **WIF or permissions fix**: You've updated a WIF provider or granted Secret Manager permissions and want NGINXaaS to retry immediately.

To reapply your configuration:

1. In the NGINXaaS console, go to your deployment.
2. Select **Reapply Configuration** in the **Configuration Info** panel.

NGINXaaS reapplies your current configuration version and immediately refetches all referenced secrets.

## Monitor secret fetch events

> NGINXaaS generates an event each time it fetches or fails to fetch a certificate or key from Secret Manager. Use these events to track successful rotations and diagnose access failures.

Reviewer: ```suggestion
NGINXaaS generates an event each time it fetches or fails to fetch a secret from Secret Manager. Use these events to track successful rotations and diagnose access failures.
```

- Select the configuration and configuration version created in the last section.
- Select **Update Configuration**.

## Rotate a Secret Manager certificate (manual)

To immediately refetch secrets without editing your NGINX configuration, use **Reapply Configuration**. This is useful in the following scenarios:

- **New secret version**: You've uploaded a new certificate and want NGINXaaS to use it right away.
- **WIF or permissions fix**: You've updated a WIF provider or granted Secret Manager permissions and want NGINXaaS to retry immediately.

To reapply your configuration:

1. In the NGINXaaS console, go to your deployment.
2. Open the **Configuration Info** panel.

> 3. Select **Reapply Configuration**.

Reviewer: What does "Open the **Configuration Info** panel" mean?

Based on the UI right now, I think we can just say "Select **Reapply Configuration** in **Configuration Info** panel".

> 3. Select **Reapply Configuration**.

Reviewer 2: I'm thinking this is missing a step:

The user either needs to update their config to reference the new Version ID, or they need to update their alias to point at the new secret version, or we are assuming they are using `latest` here but that isn't clear.

Maybe we should be better about calling out all of these approaches here as options? I guess changing their config to a new Version ID doesn't also require "Reapply Configuration" but it definitely seems like a "manual" secret manager cert rotation method.

Author: I think touching on the various scenarios is a good idea, which is why I left the above portion in: 

"This [manual rotation] is useful in the following scenarios:

- **New secret version**: You've uploaded a new certificate and want NGINXaaS to use it right away.
- **WIF or permissions fix**: You've updated a WIF provider or granted Secret Manager permissions and want NGINXaaS to retry immediately."

I don't want to complicate the sequence with more steps, as the user may choose to Reapply Config/manually rotate intentionally having made no updates.

NGINXaaS reapplies your current configuration version and immediately refetches all referenced secrets.

## Rotate a Secret Manager certificate (automatic)

> If your configuration references secrets using the `versions/latest` alias, NGINXaaS automatically picks up new certificate versions. When you add a new secret version in Secret Manager, the next scheduled sync fetches it and reloads NGINX. No configuration changes are required.

Reviewer: ```suggestion
If your configuration references secrets using `latest` as the version ID, NGINXaaS automatically picks up the latest certificate version. When you add a new secret version in Secret Manager, the next scheduled sync fetches it and reloads NGINX. No configuration changes are required.
```

Reviewer 2: another place to clarify alias functionality as well:

```suggestion
If your configuration references secrets using `latest` as the version ID, NGINXaaS automatically picks up the latest secret version on a regular cadence. When you add a new secret version in Secret Manager, the next scheduled sync fetches it and reloads NGINX.

If your configuration references secrets using a custom alias as the version ID, NGINXaaS automatically picks up the secret version referenced by that alias on a regular cadence. When you adjust the secret version an alias points to in Secret Manager, the next scheduled sync fetches that secret version and applies it to your NGINXaaS deployment.

In either of these cases, no changes to your NGINXaaS configuration are required.
```

### Rotation timeline

> New secret versions are applied to your deployment within four hours on average.

Reviewer: 4 hours is our SLO, so it shouldn't take longer than that 🤞 

```suggestion
New secret versions are applied to your deployment within four hours.
```

> To rotate your certificate:

Reviewer: "To rotate your certificate" implies this is a manual action. Maybe instead something like:

```suggestion
To ensure certificates are automatically rotated:
```

1. Generate your new certificate and key.

> 2. Add the updated certificate and key as new secret versions in Secret Manager. No configuration changes are needed.

Reviewer: Include instructions from google:

```suggestion
2. Follow Google's instructions to [add a new secret version in Secret Manager](https://docs.cloud.google.com/secret-manager/docs/add-secret-version#add-a-secret-version).
```

> 3. Wait for NGINXaaS to apply the updated certificate. Rotation is complete when the new certificate serial number appears on your NGINX endpoint and in the NGINXaaS **Certificates** list.

Reviewer: I'm not sure about saying "Wait for NGINXaaS to apply the updated certificate" as this could take a couple hours. Maybe just remove this "step"? Maybe mention that the serial number gets updated in the monitor section?

Reviewer 2: Maybe

```suggestion
3. NGINXaaS will eventually synchronize the latest certificate and key to your deployment. You can confirm the certificate(s) that are in-use by your deployment by reviewing the NGINXaaS **Certificates** list for uniquely identifying metadata, such as the serial number, or by inspecting the certificate(s) served by the specific endpoint of your deployment.
```

(maybe the last clause about inspecting the cert is overkill, or maybe it needs more detail such as an example `openssl s_client -connect -showcerts` command..)

Author: Keeping it appropriately vague:
3. When rotation is complete, confirm by checking the **Certificates** list for the new serial number, or by inspecting the certificate at your deployment's endpoint.

{{< call-out "note" >}}

> Automatic rotation only works when the Google Secret ID uses the `latest` version alias. If you've pinned a specific version number, manually rotate using **Reapply Configuration**.

Reviewer: We already have the call-out "tip" "Enable automatic rotation with latest" and the intro to this section saying the version needs to be `latest`. Do we need this here too?

> If you've pinned a specific version number, manually rotate using **Reapply Configuration**

If using a specific version number, it will not be rotated even if clicking **Reapply Configuration**. That will just fetch the same secret version.

Author: removing this block

{{< /call-out >}}

## Monitor secret fetch events

NGINXaaS generates an event each time it fetches or fails to fetch a certificate or key from Secret Manager. Use these events to track successful rotations and diagnose access failures.

### Event types

{{< table >}}
| Event type | Description |
|---|---|
| Successful Secret Fetch from Google | The secret was fetched from Secret Manager and applied to NGINX. |
| Failed Secret Fetch from Google | NGINXaaS couldn't fetch the secret. The event message includes the error details. |
{{< /table >}}

### View events in the console

- Select **Overview** in the left menu, then select **Events**. To narrow results to a specific deployment, filter by its object ID using the controls at the top of the page.
- For a summary of recent events for a specific deployment, select **Deployments**, select the deployment, and look for the **Recent Events** card. Select **See Events Details** to go to the full Events page pre-filtered for that deployment.

> ### Common failure messages and remediation

Reviewer: I like this table thanks for adding this 👍
